internal/testrunner/runners/system/system.go

```diff
@@ -5,31 +5,364 @@
+	"strings"
+	"time"
 
+	es "github.com/elastic/go-elasticsearch/v7"
+	"github.com/pkg/errors"
+
+	"github.com/elastic/elastic-package/internal/common"
+	"github.com/elastic/elastic-package/internal/elasticsearch"
+	"github.com/elastic/elastic-package/internal/install"
+	"github.com/elastic/elastic-package/internal/kibana/ingestmanager"
+	"github.com/elastic/elastic-package/internal/logger"
+	"github.com/elastic/elastic-package/internal/packages"
 	"github.com/elastic/elastic-package/internal/testrunner"
+	"github.com/elastic/elastic-package/internal/testrunner/runners/system/servicedeployer"
 )
 
+func init() {
+	testrunner.RegisterRunner(TestType, Run)
+}
+
 const (
 	// TestType defining system tests
 	TestType testrunner.TestType = "system"
 )
 
 type runner struct {
-	testFolderPath string
+	testFolder      testrunner.TestFolder
+	packageRootPath string
+	stackSettings   stackSettings
+}
+
+type stackSettings struct {
+	elasticsearch struct {
+		host     string
+		username string
+		password string
+	}
+	kibana struct {
+		host string
+	}
 }
 
 // Run runs the system tests defined under the given folder
 func Run(options testrunner.TestOptions) error {
-	r := runner{options.TestFolderPath}
+	r := runner{
+		options.TestFolder,
+		options.PackageRootPath,
+		getStackSettingsFromEnv(),
+	}
 	return r.run()
 }
 
 func (r *runner) run() error {
-	fmt.Println("system run", r.testFolderPath)
+	pkgManifest, err := packages.ReadPackageManifest(filepath.Join(r.packageRootPath, packages.PackageManifestFile))
+	if err != nil {
+		return errors.Wrap(err, "reading package manifest failed")
+	}
+
+	datasetPath, found, err := packages.FindDatasetRootForPath(r.testFolder.Path)
+	if err != nil {
+		return errors.Wrap(err, "locating dataset root failed")
+	}
+	if !found {
+		return errors.New("dataset root not found")
+	}
+
+	datasetManifest, err := packages.ReadDatasetManifest(filepath.Join(datasetPath, packages.DatasetManifestFile))
+	if err != nil {
+		return errors.Wrap(err, "reading dataset manifest failed")
+	}
+
+	// Step 1. Setup service.
+	// Step 1a. (Deferred) Tear down service.
+	logger.Info("setting up service...")
+	serviceRunner, err := servicedeployer.Factory(r.packageRootPath)
+	if err != nil {
+		return errors.Wrap(err, "could not create service runner")
+	}
+
+	tempDir, err := install.TempDir()
+	if err != nil {
+		return errors.Wrap(err, "could not get temporary folder")
+	}
+
+	ctxt := common.MapStr{}
+	ctxt.Put("service.name", r.testFolder.Package)
+	ctxt.Put("service.logs.folder.local", tempDir)
+	ctxt.Put("service.logs.folder.agent", "/tmp/service_logs/")
+
+	ctxt, err = serviceRunner.SetUp(ctxt)
+	defer func() {
+		logger.Info("tearing down service...")
+		if err := serviceRunner.TearDown(ctxt); err != nil {
+			logger.Errorf("error tearing down service: %s", err)
+		}
+	}()
+	if err != nil {
+		return errors.Wrap(err, "could not setup service")
+	}
+
+	// Step 2. Configure package (single data stream) via Ingest Manager APIs.
```

Comment:
Actually you don't use any "install" API like the one that is used in integration tests of package-registry. It looks like it's not required to be involved in the process, so I wonder what does it do actually? For reference: https://github.com/elastic/package-registry/blob/master/main_integration_test.go#L82

Maybe it would be fine to call it for safety reasons, to verify if the package can be installed?

Reply:
That's because it is being called by Agent. See https://github.com/elastic/elastic-package/pull/64/files#diff-906969eaf1b600ec57ae5e30f11d1a3dR87, https://github.com/elastic/beats/blob/2a0e0998243db56d0bfd681ff3cb06183166bdd5/dev-tools/packaging/templates/docker/docker-entrypoint.elastic-agent.tmpl#L64, and https://github.com/elastic/beats/blob/master/dev-tools/packaging/templates/docker/docker-entrypoint.elastic-agent.tmpl#L16.

Reply:
Sorry, I misguided you. I was thinking about the reference to this API method: https://github.com/elastic/package-storage/blob/production/testing/main_integration_test.go#L108

Reply:
Hmmm, good question — I wonder how I'm able to get things working even without calling this API. 🤔 Let's leave it out of this PR for now and add it if we need it in a follow up PR.